crates/driver/src/infra/simulator/tenderly/mod.rs

```diff
@@ -107,4 +107,20 @@ pub(super) enum GenerateAccessList {
 
 #[derive(Debug, Error)]
 #[error("tenderly error")]
-pub struct Error(#[from] reqwest::Error);
+pub enum Error {
+    Http(#[from] reqwest::Error),
+    Revert,
+}
+
+impl From<dto::Response> for Result<Simulation, Error> {
+    fn from(res: dto::Response) -> Self {
+        if res.transaction.status {
+            Ok(Simulation {
+                gas: res.transaction.gas_used.into(),
+                access_list: res.generated_access_list.unwrap_or_default().into(),
+            })
+        } else {
+            Err(Error::Revert)
```

Comment:
Would be nice to store some additional data in the error for debugging purposes. Maybe just storing the simulation url would already be enough.

Reply:
Will add the simulation ID to the error variant